mmeval/metrics/utils/multi_label.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+import warnings
+from typing import TYPE_CHECKING, Sequence, Tuple, Union
+
+from mmeval.utils import try_import
+
+if TYPE_CHECKING:
+    import oneflow
+    import oneflow as flow
+    import torch
+else:
+    torch = try_import('torch')
+    flow = try_import('oneflow')
+
+NUMPY_IMPL_HINTS = Tuple[Union[np.ndarray, np.number], Union[np.ndarray,
+                                                             np.number]]
+TORCH_IMPL_HINTS = Tuple['torch.Tensor', 'torch.Tensor']
+ONEFLOW_IMPL_HINTS = Tuple['oneflow.Tensor', 'oneflow.Tensor']
+BUILTIN_IMPL_HINTS = Tuple[Union[int, Sequence[Union[int, float]]],
+                           Union[int, Sequence[int]]]
+
+
+def label_to_onehot(
+        label: Union[np.ndarray, 'torch.Tensor',
+                     'oneflow.Tensor'], num_classes: int
+) -> Union[np.ndarray, 'torch.Tensor', 'oneflow.Tensor']:
+    """Convert the label-format input to one-hot encodings.
+
+    Args:
+        label (torch.Tensor or oneflow.Tensor or np.ndarray):
+            The label-format input. The format of item must be label-format.
+        num_classes (int): The number of classes.
+
+    Return:
+        torch.Tensor or oneflow.Tensor or np.ndarray:
+        The converted one-hot encodings.
+    """
+    if torch and isinstance(label, torch.Tensor):
+        label = label.long()
+        onehot = label.new_zeros((num_classes, ))
+    elif flow and isinstance(label, flow.Tensor):
+        label = label.long()
+        onehot = label.new_zeros((num_classes, ))
+    else:
+        label = label.astype(np.int64)
+        onehot = np.zeros((num_classes, ), dtype=np.int64)
+    assert label.max().item() < num_classes, \
+        'Max index is out of `num_classes` {num_classes}'
+    assert label.min().item() >= 0
+    onehot[label] = 1
+    return onehot
+
+
+def format_data(
+    data: Union[Sequence[Union[np.ndarray, 'torch.Tensor', 'oneflow.Tensor']],
+                np.ndarray, 'torch.Tensor', 'oneflow.Tensor'],
+    num_classes: int,
+    is_onehot: bool = False
+) -> Union[np.ndarray, 'torch.Tensor', 'oneflow.Tensor']:
```

Comment:
This function is slightly confusing. `is_onehot` should defaults to `None`, indicating that we have to infer. Otherwise, we should trust the given `is_onehot`, or raise an exception.

Reply:
good point, fixed